master/pkg/searcher/random.go

```diff
@@ -25,6 +25,7 @@ type (
 	// randomSearch corresponds to the standard random search method. Each random trial configuration
 	// is trained for the specified number of steps, and then validation metrics are computed.
 	randomSearch struct {
+		SearcherState
```

Comment:
1. Not sure it is a good idea to make all search methods be aware of the SearcherState.
2. More importantly, if we do this we have to use a pointer here to the same SearcherState that is used by Searcher.